libdnf/system/state.hpp

```diff
@@ -112,6 +114,11 @@ class State {
     /// @since 5.0
     void set_package_reason(const std::string & na, transaction::TransactionItemReason reason);
 
+    // @return All NA of packages installed with specified reason.
+    // @param reason The reason to search for
+    /// @since 5.0
+    std::set<std::string> get_packages_by_reason(const transaction::TransactionItemReason & reason);
```

Comment:
Because the PR does not contain the usage therefore I am fine with future change of the method.